rllm/engine/rollout/openai_engine.py

```diff
@@ -49,6 +86,45 @@ async def chat_completion(self, messages: list[dict], **kwargs) -> ModelOutput:
                     raise Exception("Rate limit reached and retries exhausted.") from None
                 print("Sleep for 5 seconds for API limit.")
                 await asyncio.sleep(5)
+            except openai.BadRequestError as e:
```

Comment:
Looks like a lot of code reuse. Can we put the logic into a function and then call it here?